tests/Unit/Configuration/WildcardToRegexTest.php

```diff
@@ -0,0 +1,29 @@
+<?php
+
+namespace DDTrace\Tests\Unit\Configuration;
+
+use DDTrace\Configuration\WildcardToRegex;
+use DDTrace\Tests\Unit\BaseTestCase;
+
+final class WildcardToRegexTest extends BaseTestCase
+{
+    /**
+     * @dataProvider wildcardToRegexExamples
+     * @param string $withWildcards
+     * @param array $regexAndReplacement
+     */
+    public function testWildcardStringsCanConvertToRegex($withWildcards, array $regexAndReplacement)
+    {
+        $this->assertSame($regexAndReplacement, WildcardToRegex::convert($withWildcards));
+    }
+
+    public function wildcardToRegexExamples()
+    {
+        return [
+            ['/foo/*/bar',                  ['|^/foo/.+/bar$|', '/foo/?/bar']],
```

Comment:
What happens to `^/foo/.+/bar$` with the pattern `/foo/123/legitimate/bar`?

Reply:
The wildcards are [greedy by default](https://github.com/DataDog/dd-trace-php/pull/442/files#diff-0ebbc6ef112d2cd18640240b6f8557a7R82). Do you think we should add a non-greedy wildcard like `*?`? :)

Reply:
Let's leave like this, and users can add their own specific paths. Also, if we receive feedback about it then we can iterate on it.